code/DDSCodeTester.cpp

```diff
@@ -3525,6 +3525,17 @@ void xml_profiles_examples()
             Subscriber* subscriber = participant->create_subscriber_with_profile("subscriber_xml_profile");
             DataReader* datareader = subscriber->create_datareader_with_profile(topic, "datareader_xml_profile");
         }
+
+        // Load XML as string data buffer
+        std::ifstream t("my_profiles.xml");
+        std::stringstream buffer;
+        buffer << t.rdbuf();
```

Comment:
Instead of loading the stream from a file, the snippet should show a string directly. My point is that if you have the file, you could already load it without putting it into a stream first. I'd create a profile
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<dds xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <profiles>
        <participant profile_name="domainparticipant_profile" is_default_profile=true>
            <domainId>2</domainId>
        </participant>
    </profiles>
</dds>
```